public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,318 @@
+                            aria-label="expression"
+                          />
+                        </Field>
+                        <Field
+                          label={
+                            <Stack gap={0.5}>
+                              <Label>Map value</Label>
+                              <Tooltip
+                                content={
+                                  <div>
+                                    <p>
+                                      Optional. Defines the name of the variable. This is currently only valid for
+                                      regular expressions with a single, unnamed capture group.
+                                    </p>
+                                  </div>
+                                }
+                              >
+                                <Icon name="info-circle" size="sm" />
+                              </Tooltip>
+                            </Stack>
+                          }
+                        >
+                          <Input
+                            {...register(`config.transformations.${index}.mapValue`)}
+                            defaultValue={fieldVal.mapValue}
+                            readOnly={readOnly}
+                            disabled={
+                              !getSupportedTransTypeDetails(watch(`config.transformations.${index}.type`)).showMapValue
+                            }
+                            aria-label="map value"
+                          />
+                        </Field>
+                        {!readOnly && (
+                          <div className={styles.removeButton}>
+                            <IconButton
+                              type="button"
+                              tooltip="Remove transformation"
+                              name={'trash-alt'}
+                              onClick={() => {
+                                remove(index);
+                                const keptValsCopy: Array<{ expression?: string; mapValue?: string } | undefined> = [
+                                  ...keptVals,
+                                ];
+                                keptValsCopy[index] = undefined;
+                                setKeptVals(compact(keptValsCopy));
+                              }}
+                              ariaLabel="Remove transformation"
+                            >
+                              Remove
+                            </IconButton>
+                          </div>
+                        )}
+                      </Stack>
+                    );
+                  })}
+                </div>
+              )}
+              {!readOnly && (
+                <Button
+                  icon="plus"
+                  onClick={() => append({ type: undefined }, { shouldFocus: false })}
+                  variant="secondary"
+                  type="button"
+                  aria-label="add transformation"
```

Comment:
you shouldn't need this, the button's content should be sufficient in labeling the button
```suggestion
```